app/assets/javascripts/oxalis/model/accessors/flycam_accessor.js

```diff
@@ -52,14 +50,25 @@ function _approximateMaxZoomForZoomStep(
   // that ~ 15 different magnifications (~ log2 of 35000) are supported properly.
   const maximumIterationCount = 100;
   let currentIterationCount = 0;
-  let maxZoomStep = 1;
-
-  while (
-    calculateTotalBucketCountForZoomLevel(dataSetScale, resolutionIndex, resolutions, maxZoomStep) <
-      maximumCapacity &&
-    currentIterationCount < maximumIterationCount
-  ) {
-    maxZoomStep *= ZOOM_STEP_INTERVAL;
+  // Since the viewports can be quite large, it can happen that even a zoom value of 1 is not feasible.
+  // That's why we start the search with a smaller value than 1. We use the ZOOM_STEP_INTERVAL factor
+  // to ensure that the calculated thresholds correspond to the normal zoom behavior.
+  let maxZoomStep = 1 / ZOOM_STEP_INTERVAL ** 20;
```

Comment:
Because you're starting 20 iterations earlier now, we only search up until 80 iterations instead of the 100 that were the maximum before (if that's understandable :see_no_evil:) Maybe we should increase `maximumIterationCount` by 20 then?

Reply:
Good point! I guess, I will rewrite this in a new branch to use a proper end criteria (which is dependent on the resolution count) as we discussed.